sos/collector/sosnode.py

```diff
@@ -838,36 +840,25 @@ def remove_file(self, path):
 
     def retrieve_sosreport(self):
         """Collect the sosreport archive from the node"""
-        if self.sos_path:
-            if self.need_sudo or self.opts.become_root:
-                try:
-                    self.make_archive_readable(self.sos_path)
-                except Exception:
-                    self.log_error('Failed to make archive readable')
-                    return False
-            self.log_info(f'Retrieving sos report from {self.address}')
-            self.ui_msg('Retrieving sos report...')
+        if self.need_sudo or self.opts.become_root:
```

Comment:
The reason why this was refactored was, where `retrieve_sosreport()` function is being used it already checks for `self.sos_path`, so the code under the else seemed redundant